pkg/virt-controller/watch/vmi.go

```diff
@@ -1056,6 +1086,21 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 	}
 
 	if !isTempPod(pod) && isPodReady(pod) {
+		if vmispec.SriovInterfaceExist(vmi.Spec.Domain.Devices.Interfaces) {
+			networkPciMapAnnotationValue, err := sriov.CreateNetworkPCIAnnotationValue(
+				vmi.Spec.Networks, vmi.Spec.Domain.Devices.Interfaces, pod.Annotations[networkv1.NetworkStatusAnnot],
+			)
+			if err != nil {
+				log.Log.V(3).Object(vmi).Warningf("failed to prepare SR-IOV network-pci map %q", err)
+			} else {
+				pod.Annotations[sriov.NetworkPCIMapAnnot] = networkPciMapAnnotationValue
+				err = c.syncPodAnnotations(pod)
+				if err != nil {
+					return &syncErrorImpl{fmt.Errorf("failed to sync pod annotations: %v", err), FailedPodPatchReason}
```

Comment:
I think this error should be close to the Patch operation that failed instead.
Or move the patch operation to be inlined "here" (at sync()).

Reply:
changed to only return the err message